backend_jobs/ingestion_pipeline/providers/image_provider_flickr.py

```diff
@@ -0,0 +1,139 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+from datetime import datetime
+import flickrapi
+from backend_jobs.ingestion_pipeline.pipeline_lib.image_provider_interface import ImageProvider
+from backend_jobs.ingestion_pipeline.pipeline_lib.data_types import ImageType
+from backend_jobs.ingestion_pipeline.pipeline_lib.data_types import ImageAttributes
+from backend_jobs.pipeline_utils import constants
+
+_FLICKR_API_KEY = '2d00397e012c30ccc33ca4fdc05a5c98'
+_FLICKR_SECRET_KEY =  'e36a277c77f09fdd'
+
+class FlickrProvider(ImageProvider):
+    """ This class is an implementation for the ImageProvider interface.
+    """
+    def __init__(self, query_arguments=None):
+        if query_arguments is not None:
+            self.query_arguments={'tag': query_arguments}
+
+    def get_images(self, num_of_page):
+        flickr = flickrapi.FlickrAPI(_FLICKR_API_KEY, _FLICKR_SECRET_KEY, cache = True)
+        photos = flickr.photos.search(text=self.query_arguments['tag'],
+                     tag_mode = 'all',
+                     tags = self.query_arguments['tag'],
+                     extras = 'url_c, geo, date_upload, date_taken, original_format,\
+                      owner_name, original_format',
+                     per_page = self.num_of_images,
+                     page = num_of_page,
+                     sort = 'relevance')
+        return photos[0]
+
+    def get_num_of_pages(self):
+        photos = self.get_images(1)
+        return int(photos.attrib['pages'])
+
+    def get_image_attributes(self, element):
+        image_arrributes=ImageAttributes(
+            url = element.get('url_c'),
+            image_id = self._generate_image_id_with_prefix(element.get('id')),
+            image_type = ImageType.CAMERA,
+            date_shot = get_date(element),
+            coordinates = get_coordinates(element),
+            format = element.get('originalformat'),
+            attribution = element.get('ownername'),
+            resolution = get_resolution(element))
+        return image_arrributes
+
+    def get_url_for_max_resolution(self, resolution, image):
+        # This function modifies the url to get the wonted resolution,
+        # in flickr the resolution is represented in the url.
+        # See details at https://www.flickr.com/services/api/misc.urls.html
+        url = image['url']
+        max_resolution =  max(resolution['height'],resolution['width'])
+        # Maping between the char that represents max resultion and the max resultion.
+        flickr_resolution_map = {75:'s',100:'t',150	:'q',240:'m',320:'n',400:'w',
+        500:'',640:'z',800:'c',1024:'b',1600:'n',2048:'k'}
+        for key in flickr_resolution_map:
+            if  max_resolution <= key:
+                if key == 500:
+                    # Removing the char that represents the resultion to get max resultion of 500.
```

Comment:
it looks like this method won't always work (some urls don't have resolution in their names according to doc). am i missing something?